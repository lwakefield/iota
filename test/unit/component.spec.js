/* eslint-env jest */
import Component from '../../src/component'

describe('Component', () => {
  it('registers components', () => {
    Component.register('foo', class Foo {})
    expect(Component.registeredComponents).toMatchSnapshot()

    const inst = new Component
    expect(inst.constructor.registeredComponents).toMatchSnapshot(0)

    // Subclasses should receive all registered comps from above
    // but the parent class should not receive components registered below
    class SubComp extends Component {}
    SubComp.register('bar', class Bar{})
    expect(SubComp.registeredComponents).toMatchSnapshot()
    expect(Component.registeredComponents).toMatchSnapshot()
  })
})
