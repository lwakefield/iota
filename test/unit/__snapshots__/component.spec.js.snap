exports[`Component registers components 1`] = `
Object {
  "foo": [Function],
}
`;

exports[`Component registers components 2`] = `
Object {
  "foo": [Function],
}
`;

exports[`Component registers components 3`] = `
Object {
  "bar": [Function],
  "foo": [Function],
}
`;

exports[`Component registers components 4`] = `
Object {
  "foo": [Function],
}
`;
